### PR TITLE
POCL: Add initial recipe.

### PR DESCRIPTION
**Pull request by maleadt**

### CODE REVIEW

File: P/pocl/bundled/patches/sysroot.patch [hunk 1]
```diff
@@ -0,0 +1,10 @@
+--- a/cmake/bitcode_rules.cmake
++++ b/cmake/bitcode_rules.cmake
+@@ -57,6 +57,7 @@
+         ${KERNEL_C_FLAGS} "-o" "${BC_FILE}" "-c" "${FULL_F_PATH}"
+         "-I${CMAKE_SOURCE_DIR}/include"
+         "-include" "${CMAKE_SOURCE_DIR}/include/_kernel_c.h"
++        "-I/opt/x86_64-linux-musl/x86_64-linux-musl/sys-root/usr/include/"
```
Comment (maleadt) [inline P/pocl/bundled/patches/sysroot.patch:7]: Where could I find the target's `math.h` instead of the host's?

Comment (giordano) [reply P/pocl/bundled/patches/sysroot.patch:7]: It should be in `/opt/${target}/${target}/sys-root/usr/include` (what you used is `/opt/${MACHTYPE}/${MACHTYPE}/sys-root/usr/include`)

Comment (maleadt) [reply P/pocl/bundled/patches/sysroot.patch:7]: Great, thanks. That does seem to break the Apple build though:

```
/opt/aarch64-apple-darwin20/aarch64-apple-darwin20/sys-root/usr/include/stdio.h:220:5: error: 'TARGET_OS_IPHONE' is not defined, evaluates to 0 [-Werror,-Wundef-prefix=TARGET_OS_]
#if TARGET_OS_IPHONE
    ^
```

IIUC, TARGET_OS_IPHONE etc are set by `TargetConditionals.h` which should be included automatically... if the directory is included via `isysroot` instead? Doing so makes it not find `math.h` anymore though. Also, AFAICT the CMake toolchain file should already make sure the sysroot is set correctly.

Comment (giordano) [reply P/pocl/bundled/patches/sysroot.patch]: What if you remove the `-Werror` flag? :grimacing:

Comment (maleadt) [reply P/pocl/bundled/patches/sysroot.patch]: Worked around this by inserting an include for `TargetConditionals.h`, but happy to hear if you have other ideas.

Comment (maleadt) [reply P/pocl/bundled/patches/sysroot.patch]: > What if you remove the `-Werror` flag? 😬

That will be tricky, as it's [inserted by Clang](https://github.com/llvm/llvm-project/blob/ef7d46c7e6b33b6294f23b6df290b335a761fbd5/clang/lib/Driver/ToolChains/Darwin.cpp#L1147-L1150)...

File: P/pocl/bundled/patches/freebsd.patch
Comment (giordano) [inline P/pocl/bundled/patches/freebsd.patch:1]: This is probably good to upstream?

Comment (maleadt) [reply P/pocl/bundled/patches/freebsd.patch:1]: https://github.com/pocl/pocl/pull/1517

**Comment by maleadt** [comment]
OK great, that only took about a day! I'm not confident the generic kernellib changes are sound, and it'd be nice to get Windows working at some point (upstream hasn't maintained it, so this will take some work). But this seems like a good starting point.

**Pull request merged**